server/server.py

```diff
@@ -0,0 +1,566 @@
+
+def update_file_in_database(client_name, file_name, file_path):    
+    """
+    Updates the database with the received file's information.
+    
+    Args:
+        client_name (str): Client's name to link the file to their account.
+        file_name (str): Name of the file.
+        file_path (str): Path where the file is saved on disk.
+    """
+    with sqlite3.connect("defensive.db") as conn_db:
+        cursor = conn_db.cursor()
+        
+        # Delete any existing record for this client and file name
+        cursor.execute('''
+            DELETE FROM files
+            WHERE ID = ? AND FileName = ?
+        ''', (clients_dict[client_name].id, file_name))
+        
+        # Insert the new file record
+        cursor.execute('''
+            INSERT INTO files (ID, FileName, PathName, Verified)
+            VALUES (?, ?, ?, ?)
+        ''', (clients_dict[client_name].id, file_name, file_path, False))
+        
+        # Update the record to Verified after the file is added
+        cursor.execute('''
+            UPDATE files
+            SET Verified = ?
+            WHERE ID = ? AND FileName = ?
+        ''', (True, clients_dict[client_name].id, file_name))
+        
+        conn_db.commit()
+
+def handle_file_transfer(conn, client_name):
+    """
+    Manages the file transfer process, including retries, decryption, and saving.
+    
+    Args:
+        conn: The connection object for receiving/sending packets.
+        client_name (str): Name of the client sending the file.
+        
+    Returns:
+        str: SUCCESS if file transfer completed; otherwise, ERROR or GENERAL_ERROR.
+    """
+    for i in range(NUMBER_OF_ATTEMPTS):
+        # Receive the initial packet and check if it is valid
+        packet = recv_packet(conn)
+        if packet['code'] != 828:
+            if i == NUMBER_OF_ATTEMPTS - 1:
+                print(f"Too many failed attempts. Incorrect packet code: {packet['code']}")
+                return ERROR
+            print(f"Error: Invalid first packet code: {packet['code']}")
+            continue
+
+        # Extract metadata for receiving the full file
+        file_name, total_packets, enc_file_size, file_size, enc_file_content = RequestUnpacker.getFileMetadata(packet)
+        print(f"Receiving file {file_name} from {client_name}, expecting {total_packets} packets.")
+
+        enc_file_content = get_enc_file_content(conn, file_name, total_packets, enc_file_size, file_size, enc_file_content)
+        if not enc_file_content:
+            if i == NUMBER_OF_ATTEMPTS - 1:
+                print("Too many failed attempts. Error in receiving content.")
+                return ERROR
+            continue
+
+        file_content = decrypt_file_content(enc_file_content, client_name, file_size)
+        if not file_content:
+            if i == NUMBER_OF_ATTEMPTS - 1:
+                print("Too many failed attempts. Error in decryption.")
+                return ERROR
+            continue

```

Comment:
yeah definitely split this loop... I can barely read this, and this continueing branches are very hard to follow...
also some of this will get better if you use exception got dammit